CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/parameters/types/FileParameterType.java

```diff
@@ -241,6 +241,7 @@ public static class FileParameterValue extends ParameterValue {
         private ExtensionFilter filter;
         private boolean acceptAllFileFilterUsed;
         private boolean warnOverwrite;
+        private boolean fileChooserSelected;
```

Comment:
Instead of a boolean, could we instead hold a list of filenames that the user has clicked "Yes" to overwrite the file.
Using the file chooser would always add an entry to the list (even if it's not overwriting an existing file).
Manually modifying the filename afterwards should check if the new filename is in the list ... if not, check if the filename matches an existing file on the machine and if so then display the overwrite warning. 
If the user clicks on "yes" to overwrite, then add that filename to the list.
Note it should not clear the list if there's an error in the path while the user is typing a new filename.